src/pip/_internal/index/collector.py

```diff
@@ -448,7 +448,7 @@ def _get_index_content(
 ) -> Optional["IndexContent"]:
     if session is None:
         raise TypeError(
-            "_get_html_page() missing 1 required keyword argument: 'session'"
+            "_get_index_content() missing 1 required keyword argument: 'session'"
         )
```

Comment:
Actually, since we have static types, we could probably remove this test and remove the Optional in the argument type.
But I'm happy to merge this as is.

Reply:
This is from Python 2 days — we should change this to be a required kwarg now that we’re Python 3.

But, yea, in a follow up.

Reply:
I totally agree but there is an issue in your comment : the word "probably".
Reason why I did not go for removal.